docs/v3.x/getting-started/react.md

```diff
@@ -0,0 +1,577 @@
+
+::: tab axios
+
+*Request*
+
+```js
+axios.post('http://localhost:1337/restaurants', {
+    name: 'Dolemon Sushi',
+    description: 'Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious'
+  })
+  .then(response => {
+    console.log(response);
+  })
+  .catch(error => {
+    console.log(error);
+  });
+```
+
+:::
+
+::: tab fetch
+
+*Request*
+
+```js
+fetch('http://localhost:1337/restaurants', {
+   method: 'POST',
+   headers: {
+      'Content-Type': 'application/json'
+   },
+   body: JSON.stringify({
+     name: 'Dolemon Sushi',
+     description: 'Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious'
+   })
+ }).then(response => {
+   return response.json();
+ }).then(data => {
+   console.log(data);
+ }).catch(error => {
+   console.error(error);
+ });
+```
+
+:::
+
+*Response*
+
+```json
+{
+    "id": 2,
+    "name": "Dolemon Sushi",
+    "description": "Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious",
+    "created_by": null,
+    "updated_by": null,
+    "created_at": "2020-08-04T09:57:11.669Z",
+    "updated_at": "2020-08-04T09:57:11.669Z",
+    "categories": [
+    ]
+}
+```
+
+::::
+
+### Example
+
+:::: tabs
+
+::: tab axios
+
+`./src/App.js`
+
+```js
+import React from 'react';
+import axios from 'axios'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      description: '',
+      categories: '',
+      allCategories: [],
+      error: null
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:1337/categories')
+    .then(response => {
+      const categories = response.data;
+      this.setState({ allCategories: categories });
+    })
+    .catch(error => {
+      this.setState({ error });
+    })
+  }
+
+  handleInputChange(event) {
+     const target = event.target;
+     const value = target.value;
+     const name = target.name;
+
+     this.setState({
+       [name]: value
+     });
+   }
+
+  handleSubmit = (e) =>  {
+    e.preventDefault();
+
+    axios.post('http://localhost:1337/restaurants', this.state)
+      .then(response => {
+        console.log(response);
+      })
+      .catch(error => {
+        this.setState({ error });
+      });
+  }
+
+  render() {
+    const { error, categories, allCategories } = this.state
+    if (error) {
+       return <div>An error occured: {error.message}</div>
+    }
+    return (
+      <div className="App">
+        <form onSubmit={this.handleSubmit}>
+           <label>
+             Name:
+             <input type="text" name="name" onChange={this.handleInputChange} />
+           </label>
+           <label>
+             Description:
+             <input type="text" name="description" onChange={this.handleInputChange} />
+           </label>
+           <select name="categories" value={categories} onChange={this.handleInputChange}>
```

Comment:
A select input is not the best DOM element for your use case. Here's my suggestion:

```js
render() {
  return (
      // ...
      <label>
            Description:
            <input
              type="text"
              name="description"
              onChange={this.handleInputChange}
              value={modifiedData.description}
            />
          </label>
          <div>{allCategories.map(this.renderCheckbox)}</div>
          // ... 
  )
}
```

Then the `renderCheckbox` instance should be something like this:

```js
renderCheckbox = category => {
    const {
      modifiedData: { categories }
    } = this.state;
    const isChecked = categories.includes(category.id);

    const handleChange = () => {
     // We need to add the category
      if (!categories.includes(category.id)) {
        this.handleInputChange({
          target: { name: "categories", value: categories.concat(category.id) }
        });

     // Remove the category
      } else {
        this.handleInputChange({
          target: {
            name: "categories",
            value: categories.filter(v => v !== category.id)
          }
        });
      }
    };
```